pkg/gardenlet/controller/shoot/shoot/helper/infos.go

```diff
@@ -108,22 +122,33 @@ func CalculateControllerInfos(shoot *gardencorev1beta1.Shoot, clock clock.Clock,
 	return i
 }
 
-func (i ControllerInfos) shouldReconcileNow() bool {
+func (i ControllerInfos) shouldReconcileNow() ReconcileNowDecision {
 	// if the shoot is failed or ignored, it doesn't matter which operation is triggered
 	if i.isFailed || i.isIgnored {
-		return false
+		return ReconcileNowDecision{
+			Result: false,
+			Reason: "Shoot is failed or ignored.",
+		}
+	}
+
+	// Emergency switch: Check Seed annotation to block reconciliation if needed
+	if i.emergencyStopReconciliations {
+		return ReconcileNowDecision{
+			Result: false,
+			Reason: "Shoot reconciliation blocked by Seed emergency switch",
+		}
```

Comment:
Nit: I would move this up at the very top. This helps reporting the same message for failed or ignored shoots :)